vignettes/use-application-audit.Rmd

```diff
@@ -0,0 +1,492 @@
+
+``` r
+record_loadtest <- function(path, timeout = 15, workers = 5) {
+  message("\n---- BEGIN LOAD-TEST ---- \n")
+  # start app + recorder
+  target <- start_r_bg(shiny_bg, path = path)
+  recorder <- start_r_bg(recorder_bg)
+
+  # start headless chrome (points to recorder!).
+  # AppDriver also support remote urls.
+  chrome <- shinytest2::AppDriver$new(
+    "http://127.0.0.1:8600",
+    load_timeout = timeout * 1000
+  )
+
+  chrome$set_inputs(mu = 4, timeout_ = timeout * 1000)
+
+  # clean
+  chrome$stop()
+  # needed to avoid
+  # java.lang.IllegalStateException: last event in log not a
+  # WS_CLOSE (did you close the tab after recording?)
+  Sys.sleep(2)
+
+  # shinycannon (maybe expose other params later ...)
+  target_url <- "http://127.0.0.1:3515"
+  system(
+    sprintf(
+      "shinycannon recording.log %s --workers %s --loaded-duration-minutes 2 --output-dir run1",
+      target_url, workers
+    )
+  )
+
+  target$kill()
+
+  # Treat data and generate report
+  df <- shinyloadtest::load_runs("run1")
+  shinyloadtest::shinyloadtest_report(
+    df,
+    "public/index.html",
+    self_contained = TRUE,
+    open_browser = FALSE
+  )
+}
+```
+
+Below is the necessary GitHub Actions yaml file. Overall, this will:
+
+- Run each time code is pushed to GitHub.
+- Run in parallel on 3 different Ubuntu flavors.
+- Install R, some system dependencies and R packages. Install Chrome and 
+shinycannon tools for the headless loadtest.
+- Deploy the load test report to GitHub Pages.
+
+```yaml
+on:
+  push:
+    branches: [main, master]
+  pull_request:
+    branches: [main, master]
+
+name: shiny-loadtest-ci
+
+jobs:
+  shiny-loadtest-ci:
+    runs-on: ${{ matrix.config.os }}
+    permissions:
+      contents: write
+
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: ubuntu-latest,   r: 'devel', http-user-agent: 'release'}
+          - {os: ubuntu-latest,   r: 'release'}
+          - {os: ubuntu-latest,   r: 'oldrel-1'}
+
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: r-lib/actions/setup-r@v2
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          cache-version: 2
+          extra-packages: |
+            any::shinyloadtest
+            any::lubridate
+            any::DT
+            any::callr
+            any::shinytest2
+            any::deSolve
+            any::attempt
+
+      - name: Install shinycannon 💥 and Chrome 🌐
+        run: |
+          sudo bash -c 'apt-get update; apt-get install -y default-jre-headless; apt-get clean; rm -rf /var/lib/apt/lists/*'
+          wget https://github.com/rstudio/shinycannon/releases/download/v1.1.3/shinycannon_1.1.3-dd43f6b_amd64.deb https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
```

Comment:
Chrome is handled in pak installation in prior step. Removing chrome install. Keeping shinycannon install

```suggestion
          wget https://github.com/rstudio/shinycannon/releases/download/v1.1.3/shinycannon_1.1.3-dd43f6b_amd64.deb
```